File: Interpreter/Interpreter/Models/serialize/RepositoryModel.cs
using System.Collections.Generic;
using Interpreter.Models.complexStatementTypes;

namespace Interpreter.Models{
    public class RepositoryModel : IStatement
    {
            public VcxprojStructure vcxprojStructure { get; set; }

            public IList<ClassOrHeaderWithPath> listOfHeaderFiles { get; set; } = new List<ClassOrHeaderWithPath>();

            public IList<ClassOrHeaderWithPath> listOfCppFiles { get; set; } = new List<ClassOrHeaderWithPath>();
    }
};

File: src/main/kotlin/com/stannis/dataModel/complexStatementTypes/RepositoryModel.kt
package com.stannis.dataModel.complexStatementTypes

import com.stannis.dataModel.Statement
import com.stannis.parser.sln.VcxprojStructure
import java.util.StringJoiner

data class RepositoryModel(
    override val `$type`: String? = "RepositoryModel",
    val listOfHeaderFiles: ArrayList<ClassOrHeaderWithPath>?,
    val listOfCppFiles: ArrayList<ClassOrHeaderWithPath>?,
    val vcxprojStructure: VcxprojStructure
): Statement
